### ISSUE TITLE
SimCity 4 Known Instance TGIs

### ISSUE THREAD

**Issue by chances** [anonhostpi/DBPF.js#3]
Regarding DBPF archive file handlers, would you accept a PR that adds something like my list of SC4-related [`KnownInstance`](https://github.com/chances/dbpf-d/blob/07005a3aa93a29be8d7853b726543ec4a0a0596d/src/dbpf/tgi.d#L48) TGIs?

Would I also modify the list of [Resource Types](https://anonhostpi.github.io/DBPF.js/docs/other/spec/sections/ResourceTypes)?

**Comment by anonhostpi** [comment]
> Would I also modify the list of [Resource Types](https://anonhostpi.github.io/DBPF.js/docs/other/spec/sections/ResourceTypes)?

Yes, the scripts/resources.js automatically parses the ResourceTypes.md file to generate basic plugin files like details.json and any KaiTai structs.
- The KaiTai part is still in the works, as I have to implement the KaiTai-to-JS compiler in my build files.
- I also have to draft up/work on some system/way for converting the KaiTai build output to TypeScript in a way that can be scaled. For now, I'll be writing the type defs manually.

NOTE: my scripts/resources.js is shottily written. It's currently written to get the job done, but I will eventually find a way to clean it up.

**Comment by anonhostpi** [comment]
As for PR, I might not accept it right off the bat, but that's NOT because I'm against contributions.

More than likely, I will have to make adjustments to your fork after I iron things out like KaiTai compilation and multithreading.

I would definitely start the fork.

Then, we'll look at it to see what changes need to be made to support the multithreading and KaiTai changes I'm working on locally
